### PR TITLE
Fast Front-ends: Check for modals and notification query params client side

### PR DESCRIPTION
**Pull request by hamstu**
## Description

Logic is the same, just moving the checks client side. For simplicity still putting the modal and notification data into global vars as it felt more lightweight than creating an entire new store/reducer.

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/modals/middleware.js [hunk 1]
```diff
@@ -18,6 +18,20 @@ import {
 export default ({ dispatch, getState }) => next => action => {
   next(action);
   switch (action.type) {
+    case 'APP_INIT': {
+      let modal = null;
```
Comment (albennett) [inline packages/modals/middleware.js:22]: I was curious about how you were going to handle the modal key params. 😄 

We rely quite a lot on requests on APP_INIT. As the list grows, we could question whether all the requests need to be initially called, and whether some can be called after we get key data for visual purposes. Though I may be misunderstanding something! 😄

File: packages/thirdParty/middleware.js [hunk 2]
```diff
@@ -10,16 +10,6 @@ import * as FullStory from '@fullstory/browser';
 
 import { actionTypes } from './reducer';
 
-const checkExtensionInstalled = () => {
```
Comment (albennett) [inline packages/thirdParty/middleware.js:13]: Hm, is this no longer needed?

File: packages/server/lib/generateIndexHtml.js [hunk 3]
```diff
@@ -13,14 +11,7 @@ const getDatadogRumScript = require('./embeds/datadogRum');
...
 
-const getHtml = ({
-  staticAssets,
-  isProduction,
-  isStandalone,
-  notification,
-  modalKey,
-  modalValue,
-}) => {
+const getHtml = ({ staticAssets, isProduction, isStandalone }) => {
```
Comment (albennett) [inline packages/server/lib/generateIndexHtml.js:14]: 😍

**Review by albennett — APPROVED** [review]
Great job @hamstu 🙌 Before long, allll the scripts will be gone.

File: packages/thirdParty/middleware.js (hunk 2 above)
Comment (hamstu) [reply packages/thirdParty/middleware.js:13]: Yeah it's not used any more 👍

File: packages/modals/middleware.js (hunk 1 above)
Comment (hamstu) [reply packages/modals/middleware.js:22]: Good one! The reason I use APP_INIT is that we might trigger navigation / redirects once the React app renders, so we need to pull the query strings first, before that (potentially) happens.

We could also dispatch another event that comes early, but it felt simple enough to piggyback on `APP_INIT` for this one. Let me know if that makes sense!

**Comment by hamstu** [comment]
Thank you @albennett! And yes, we're getting close to a simplified experience here I hope! 😄

**Pull request merged**